django/template/base.py

```diff
@@ -131,7 +131,15 @@ class TemplateSyntaxError(Exception):
 
 
 class TemplateDoesNotExist(Exception):
-    pass
+    """
+    This exception is used when template loaders are unable to find a
+    template. The tried argument is an optional list of tuples containing
+    (origin, status), where origin is an Origin object and status is a string
+    with the reason the template wasn't found.
+    """
+    def __init__(self, msg, tried=None):
+        self.tried = tried or []
+        super(TemplateDoesNotExist, self).__init__(msg)
```

Comment:
Bikeshedding: I think ``TemplateDoesNotExist`` makes more sense in ``django.template.loader`` than ``django.template.base``. We can't move it yet, though, until ``django.template.loader.BaseLoader`` is removed. Otherwise it leads to circular imports.